From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: anzz1 <anzz1@live.com>
Date: Wed, 29 Jul 2026 15:40:47 +0000
Subject: [PATCH] Fix MC-233911 Wakeup Time Sync

Sends a time sync packet immediately after waking up from sleep.

diff --git a/src/main/java/net/minecraft/server/WorldServer.java b/src/main/java/net/minecraft/server/WorldServer.java
index f148c6931fb2b935adea6deb6d909ab596d5030c..b05baecc5180bf8506da30142ef24816975b0d50 100644
--- a/src/main/java/net/minecraft/server/WorldServer.java
+++ b/src/main/java/net/minecraft/server/WorldServer.java
@@ -212,11 +212,13 @@ public class WorldServer extends World /*implements IAsyncTaskHandler*/ { // Pan
         }
 
         this.worldProvider.m().b();
+        boolean timeNeedsSync = false; // PandaSpigot - Wakeup Time Sync
         if (this.everyoneDeeplySleeping()) {
             if (this.getGameRules().getBoolean("doDaylightCycle")) {
                 long i = this.worldData.getDayTime() + 24000L;
 
                 this.worldData.setDayTime(i - i % 24000L);
+                timeNeedsSync = true; // PandaSpigot - Wakeup Time Sync
             }
 
             this.e();
@@ -243,6 +245,9 @@ public class WorldServer extends World /*implements IAsyncTaskHandler*/ { // Pan
         this.worldData.setTime(this.worldData.getTime() + 1L);
         if (this.getGameRules().getBoolean("doDaylightCycle")) {
             this.worldData.setDayTime(this.worldData.getDayTime() + 1L);
+            if (timeNeedsSync) {
+                this.sendWorldTimeUpdate(true); // PandaSpigot - Wakeup Time Sync
+            }
         }
 
         timings.doChunkUnload.stopTiming(); // Spigot
